### PR TITLE
[Catalog] Fixes improper use of Bundle and assets

### LINKED ISSUE #1941: Fix incorrect use of NSBundle
A common pattern in our code is:

```
// Bad
 NSBundle *bundle = [NSBundle bundleForClass:[self class]];
```

Per @brianjmoore in #1930 this will fail for subclasses.  We should be explicit about the class as in:

```
 NSBundle *bundle = [NSBundle bundleForClass:[MDCActivityIndicator class]];
```

Fix incorrect use throughout codebase.

### PR DESCRIPTION
**Pull request by willlarche**
Closes #1941 

If it looks strange formatting-wise, that's what the clang formatter wants done!

### CODE REVIEW

File: components/AppBar/examples/AppBarImageryExample.m [hunk 1]
```diff
@@ -28,7 +28,11 @@ - (void)viewDidLoad {
   [super viewDidLoad];
 
   // Create our custom image view and add it to the header view.
-  UIImageView *imageView = [[UIImageView alloc] initWithImage:[self headerBackgroundImage]];
+  UIImageView *imageView = [[UIImageView alloc]
```
Comment (romoore) [inline components/AppBar/examples/AppBarImageryExample.m:31]: Super-nit: two-liner?

```
UIImage ...
UIImageView ...
```

Comment (willlarche) [reply components/AppBar/examples/AppBarImageryExample.m:31]: Clang-format!

File: components/HeaderStackView/examples/supplemental/HeaderStackViewTypicalUseSupplemental.m [hunk 2]
```diff
@@ -44,7 +44,12 @@ - (void)setupExampleViews {
   self.topView.autoresizingMask =
       UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
 
-  UIImageView *imageView = [[UIImageView alloc] initWithImage:[self headerBackgroundImage]];
+  UIImageView *imageView = [[UIImageView alloc]
```
Comment (romoore) [inline components/HeaderStackView/examples/supplemental/HeaderStackViewTypicalUseSupplemental.m:47]: same super-duper-nit

File: components/AppBar/examples/AppBarImageryExample.m (hunk 1 above)
Comment (willlarche) [reply components/AppBar/examples/AppBarImageryExample.m:31]: It's actually a super long line and it's breaking it up at each opportunity.

**Review by romoore — APPROVED** [review]

File: components/AppBar/examples/AppBarImageryExample.m (hunk 1 above)
Comment (romoore) [reply components/AppBar/examples/AppBarImageryExample.m:31]: Yah, I know. Robots will be robots...  Feel free not to fix. 👍

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
Please merge when Travis is happy.

**Pull request merged**